### ISSUE TITLE
KMM on iOS looks for UpdateMode crashing the build

### ISSUE THREAD

**Issue by tiwiz** [Kotlin/kotlinx.serialization#1162]
Whenever I update the library to the stable version 1.0.0, I get this error that crashes my iOS build (it still works find on Android): 
```
e: java.lang.IllegalStateException: Deserializer for declaration public kotlinx.serialization/UpdateMode|null[0] is not found
```

The full stack trace can be found [here](https://gist.github.com/tiwiz/670807f5d93458c458128e89ec7b0ccc)

RIght now, the issue can be solved by [reverting to version 1.0.0-RC](https://github.com/tiwiz/esb_lights_kmm/commit/206b2a54629ceb036b30671154475b57e700e50b) where `UpdateMode` was still there

**To Reproduce**
I experienced this bug on my current project (https://github.com/tiwiz/esb_lights_kmm/blob/master/shared/build.gradle.kts). To reproduce, increase the `kotlinx.serialization` version to 1.0.0

**Expected behavior**
I would expect version 1.0.0 to compile as 1.0.0-RC

**Environment**
 - Kotlin version: 1.4.0
 - Library version: 1.0.0
 - Kotlin platforms: JVM, Native
 - Gradle version: 6.7

**Comment by sandwwraith** [comment]
It looks like you have transitive dependency that still uses 1.0.0-RC, therefore you can't update to 1.0.0

**Comment by sandwwraith** [comment]
See also #1110

**Comment by tiwiz** [comment]
It might be Ktor? Should I open an issue on the Ktor repo to ask for a dependency upgrade?

**Comment by jarroyoesp** [comment]
I am having the same issue. Does anybody know how to solve it?

**Comment by qwwdfsad** [comment]
Should be fixed since 1.2.0
